File: spring-boot-java-13/src/main/java/de/rieckpil/test/springbootjava13/SampleController.java
package de.rieckpil.test.springbootjava13;

import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/json")
public class SampleController {

  @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
  public String getJson() {
    return """
          {
            "name":"duke",
            "age": 42
          }
      """;
  }
}

File: webclient-stacktrace/src/main/kotlin/de/rieckpil/learning/webclientstacktrace/SampleController.kt
package de.rieckpil.learning.webclientstacktrace

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api")
class SampleController(private val sampleClient: SampleClient) {

    @GetMapping
    fun getSampleData() = sampleClient.getData()

}